x-pack/plugins/security_solution/server/endpoint/routes/actions/isolation.ts

```diff
@@ -52,6 +60,31 @@ export function registerHostIsolationRoutes(
   );
 }
 
+const doLogsEndpointActionDsExists = async ({
+  esClient,
+  logger,
+  dataStreamName,
+}: {
+  esClient: ElasticsearchClient;
+  logger: Logger;
+  dataStreamName: string;
+}): Promise<boolean> => {
+  let doesIndexTemplateExist;
+  try {
+    doesIndexTemplateExist = await esClient.indices.existsIndexTemplate({
+      name: dataStreamName,
+    });
+  } catch (error) {
+    const errorType = error?.type ?? '';
+    if (errorType !== 'resource_not_found_exception') {
+      logger.error(error);
+      throw error;
+    }
+    return false;
+  }
+  return doesIndexTemplateExist.statusCode === 404 ? false : true;
```

Comment:
nit: you can put this return inside the try 

```
try {
    const doesIndexTemplateExist = await esClient.indices.existsIndexTemplate({
      name: dataStreamName,
    });
    return doesIndexTemplateExist.statusCode === 404 ? false : true;
  } catch (error) {
    const errorType = error?.type ?? '';
    if (errorType !== 'resource_not_found_exception') {
      logger.error(error);
      throw error;
    }
    return false;
  }
```

Reply:
I wanted to have a default `return`, but yeah I see this is a bit simpler to reason with.

Reply:
e1e3d4e2fc123b1a2dcba7f84dbe7ad23c4bdbff